examples/plot_bch_so3.py

```diff
@@ -0,0 +1,93 @@
+"""
+Visualize the first, second and third order approximation of the Baker Campbell
+Hausdorf formula on so(n). To this end, sample 2 random elements a,b of so(n)
+and compute both the BCH approximations of different orders as well as
+log(exp(a)exp(b)) and compare these in the Frobenius norm.
+
+Notice that the BCH only guarantees convergence if ||a|| + ||b|| < log 2,
+so we normalize the random vectors to have norm 1 / 2.
+
+We also compare execution times of the scikit-learn expm / logm implementation
+with our BCH-implementation, for small orders approximation by BCH is faster
+than the scikit-learn version, while being close to the actual value.
+
+"""
+import timeit
+
+import matplotlib.pyplot as plt
+
+import geomstats.backend as gs
+from geomstats.geometry.skew_symmetric_matrices import SkewSymmetricMatrices
+from geomstats.geometry.special_orthogonal_group import SpecialOrthogonalGroup
+
+
+n = 3
+max_order = 10
+
+group = SpecialOrthogonalGroup(n=n)
+group.default_point_type = "matrix"
+
+dim = int(n * (n - 1) / 2)
+algebra = SkewSymmetricMatrices(n=n)
+
+
+def main():
+    norm_rv_1 = gs.normal(size=dim)
+    tan_rv_1 = algebra.matrix_representation(
+        norm_rv_1 / gs.norm(norm_rv_1, axis=0) / 2
+    )
```

Comment:
NIT: We haven't coded with single parenthesis on oneline so far. If you feel motivated, bringing the ) one line will make the code more consistent